routing/routing.go

```diff
@@ -238,7 +238,8 @@ type PreProcessor interface {
 // Routing ('router') instance providing live
 // updatable request matching.
 type Routing struct {
-	routeTable        atomic.Value // of struct routeTable
```

Comment:
when this atomic value was introduce, there was an actual performance improvement in the route lookup that made it worth it. Since then, many things have changed, including the Go runtime scheduler. Can we run a benchmark comparison before and after this change?

Reply:
found it! Checking... :smile:

Reply:
Nice! :+1:

```
BenchmarkRoutingGetAtomicValue/goroutines-1200-4         	45787898	        25.7 ns/op
BenchmarkRoutingGetAtomicValue/goroutines-1400-4         	54809168	        25.2 ns/op
BenchmarkRoutingGetAtomicValue/goroutines-1600-4         	43664787	        26.6 ns/op
BenchmarkRoutingGetAtomicValue/goroutines-1800-4         	43672563	        26.6 ns/op
BenchmarkRoutingGetAtomicValue/goroutines-2000-4         	41791792	        25.8 ns/op
BenchmarkRoutingGetAtomicValue/goroutines-2200-4         	45046342	        24.3 ns/op
BenchmarkRoutingGetAtomicValue/goroutines-2400-4         	45706029	        25.7 ns/op
BenchmarkRoutingGetAtomicValue/goroutines-2600-4         	40851520	        26.3 ns/op
BenchmarkRoutingGetMutex/goroutines-1200-4               	 9082144	       132 ns/op
BenchmarkRoutingGetMutex/goroutines-1400-4               	 9578566	       129 ns/op
BenchmarkRoutingGetMutex/goroutines-1600-4               	10937642	       132 ns/op
BenchmarkRoutingGetMutex/goroutines-1800-4               	16323981	       129 ns/op
BenchmarkRoutingGetMutex/goroutines-2000-4               	 9657950	       127 ns/op
BenchmarkRoutingGetMutex/goroutines-2200-4               	10629856	       130 ns/op
BenchmarkRoutingGetMutex/goroutines-2400-4               	12391550	       126 ns/op
BenchmarkRoutingGetMutex/goroutines-2600-4               	10572556	       123 ns/op
BenchmarkRoutingGetRWMutex/goroutines-1200-4             	17687150	        67.4 ns/op
BenchmarkRoutingGetRWMutex/goroutines-1400-4             	17627328	        66.8 ns/op
BenchmarkRoutingGetRWMutex/goroutines-1600-4             	17076651	        68.0 ns/op
BenchmarkRoutingGetRWMutex/goroutines-1800-4             	16905411	        66.4 ns/op
BenchmarkRoutingGetRWMutex/goroutines-2000-4             	16989332	        66.5 ns/op
BenchmarkRoutingGetRWMutex/goroutines-2200-4             	17167993	        66.4 ns/op
BenchmarkRoutingGetRWMutex/goroutines-2400-4             	17821053	        65.1 ns/op
BenchmarkRoutingGetRWMutex/goroutines-2600-4             	17468914	        67.1 ns/op
```

Reply:
Yeah, with this ns/op values it does not look as a big deal to me